source/sessions/tests/README.rst

```diff
@@ -234,6 +233,39 @@ This test only applies to drivers that allow authentication to be changed on the
 * Call ``findOne`` using the session as an explicit session
 * Assert that the driver returned an error because the session is owned by a different user
 
+18. Implicit session is ignored if connection does not support sessions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This test uses a mongocryptd server as the test server; however, in the event that mongocryptd 
+starts supporting sessions or is not a viable option for this test, another server may be substituted
+in these tests as long as it does not return a non-null value for ``logicalSessionTimeoutMinutes``;
+in the event that no such server is readily available, a mock server may be used instead. 
+
+* Ensure a mongocryptd process is running
+* Create a new ``MongoClient`` pointed at the mongocryptd server with command monitoring enabled
+* Verify that the server does NOT define a value for ``logicalSessionTimeoutMinutes`` by sending a hello command to the server and checking the response
+* Send a read command to the server (e.g., ``findOne``), ignoring any errors from the server response
+* Check the corresponding ``commandStarted`` event: verify that ``lsid`` is not set
+* Send a write command to the server (e.g., ``insertOne``), ignoring any errors from the server response
+* Check the corresponding ``commandStarted`` event: verify that lsid is not set
+
+19. Explicit session raises an error if connection does not support sessions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This test uses a mongocryptd server as the test server; however, in the event that mongocryptd 
+starts supporting sessions or is not a viable option for this test, another server may be substituted
+in these tests as long as it does not return a non-null value for ``logicalSessionTimeoutMinutes``;
+in the event that no such server is readily available, a mock server may be used instead. 
+
+* Ensure a mongocryptd process is running
+* Create a new ``MongoClient`` pointed at the mongocryptd server
+* Verify that the server does NOT define a value for ``logicalSessionTimeoutMinutes`` by sending a hello command to the server and checking the response
+* Create a new explicit session by calling ``startSession`` (this MUST NOT error)
+* Attempt to send a read command to the server (e.g., ``findOne``) with the explicit session passed in
+* Assert that a client-side error is generated indicating that sessions are not supported
+* Attempt to send a write command to the server (e.g., ``insertOne``) with the explicit session passed in
+* Assert that a client-side error is generated indicating that sessions are not supported
```

Comment:
Should we combine these tests to avoid the duplication? I imagine when I implement this in python I would make it one test anyway:
```
...
* Send a read command to the server (e.g., ``findOne``), ignoring any errors from the server response
* Check the corresponding ``commandStarted`` event: verify that ``lsid`` is not set
* Send a write command to the server (e.g., ``insertOne``), ignoring any errors from the server response
* Check the corresponding ``commandStarted`` event: verify that lsid is not set
* Create a new explicit session by calling ``startSession`` (this MUST NOT error)
* Attempt to send a read command to the server (e.g., ``findOne``) with the explicit session passed in
* Assert that a client-side error is generated indicating that sessions are not supported
* Attempt to send a write command to the server (e.g., ``insertOne``) with the explicit session passed in
* Assert that a client-side error is generated indicating that sessions are not supported
```

Reply:
Code reuse is an implementation detail for the individual driver. There are two distinct scenarios being tested, so from the perspective of clarity of intent, it makes sense to have two distinct test cases (even if some of the setup is shared). The implicit session scenario could fail while the explicit session scenario is passing and vice versa. If a driver ever has a bug which respect to one of these conditions, there is benefit to having one of the tests implemented and passing and one skipped versus having a combined test skipped or a combined test running where the description doesn't match the assertions because some of them must be skipped until a fix is implemented.

Reply:
I agree it's up to the implementor wether to combine or not but I would still prefer to avoid duplicating the test setup part in the description (with the aim of making these tests easier to read/maintain). If there's no immediately obvious way to do that then I'm happy as is.

Reply:
I did think about combining the note about mongocryptd setup: if you think it would improve readability, I could factor this out as a note about "Testing against servers that do not support sessions" and then reference it in both test cases

Reply:
Made the corresponding update, let me know how that looks